node/src/main/scala/coop/rchain/node/balance/Transaction.scala

```diff
@@ -0,0 +1,147 @@
+                            amount        <- j.hcursor.downField("amount").as[Long]
+                            retUnforeable <- j.hcursor.downField("retUnforeable").as[String]
+                            success       <- j.hcursor.downField("success").as[Boolean]
+                            reason        <- j.hcursor.downField("reason").as[String]
+                          } yield TransactionInfo(
+                            fromAddr = fromAddr,
+                            toAddr = toAddr,
+                            amount = amount,
+                            retUnforeable = retUnforeable,
+                            deploy = d,
+                            success = success,
+                            reason = reason
+                          )
+                      )
+                      .toList
+                      .traverse(identity)
+                  }
+                })
+                .toList
+                .traverse(identity)
+          }
+    }
+
+  def parseTransactionInfoJson(rawJson: String): Result[List[List[TransactionInfo]]] = {
+    val parsed = parse(rawJson).getOrElse(Json.Null)
+    val cursor = parsed.hcursor
+    decodeTransactionInfo(cursor)
+  }
+
+  def storeManager[F[_]: Concurrent: Log](dirPath: Path): F[KeyValueStoreManager[F]] =
+    LmdbStoreManager(dirPath, 1024L * 1024L * 1024L)
+
+  def store[F[_]: Concurrent: Log](
+      dirPath: Path
+  ): F[TransactionStore[F]] =
+    LmdbStoreManager(dirPath, 1024L * 1024L * 1024L) >>= { manager =>
+      manager
+        .store("transaction")
+        .map(_.toTypedStore(transactionKeyCodec, transactionValueCodec))
+    }
+
+  val transactionKeyCodec = new Codec[String] {
+    override def encode(value: String): Attempt[BitVector] =
+      Attempt.successful(BitVector(value.getBytes("UTF8")))
+
+    override def decode(bits: BitVector): Attempt[DecodeResult[String]] =
+      Attempt.successful(
+        DecodeResult(StandardCharsets.UTF_8.decode(bits.toByteBuffer).toString, BitVector.empty)
+      )
+
+    override def sizeBound: SizeBound = SizeBound.unknown
+  }
+
+  val transactionValueCodec = new Codec[List[List[TransactionInfo]]] {
+    override def encode(value: List[List[TransactionInfo]]): Attempt[BitVector] =
+      Attempt.successful(BitVector(value.asJson.toString().getBytes(StandardCharsets.UTF_8)))
+
+    override def decode(bits: BitVector): Attempt[DecodeResult[List[List[TransactionInfo]]]] = {
+      val value = StandardCharsets.UTF_8.decode(bits.toByteBuffer).toString
+      val target = if (value == "[]") {
+        "[[]]"
+      } else value
+      val result = parseTransactionInfoJson(target)
+      result match {
+        case Right(value) => Attempt.successful(DecodeResult(value, BitVector.empty))
+        case Left(f)      => Attempt.failure(Err.General(f.message, List.empty))
+      }
+    }
+
+    override def sizeBound: SizeBound = SizeBound.unknown
+  }
```

Comment:
I'm curious why you didn't use automatic derivation like in Web API?
https://circe.github.io/circe/codecs/auto-derivation.html